math/miller_rabin.cpp

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright 2020 @author tjgurwara99
+ * @file
+ *
+ * A basic implementation of Miller-Rabin primality test.
+ */
+
+#include <cassert>
+#include <iostream>
+#include <random>
+#include <vector>
+
+/**
+ * Function to give a binary representation of a number in reverse order
+ * @param num integer number that we want to convert
+ * @return result vector of the number input in reverse binary
+ */
+template <typename T> std::vector<T> reverse_binary(T num) {
+    std::vector<T> result;
+    T temp = num;
+    while (temp > 0) {
+        result.push_back(temp % 2);
+        temp = temp / 2;
+    }
+    return result;
+}
+
+/**
+ * Function for modular exponentiation.
+ * This function is an efficient modular exponentation function.
+ * It can be used with any big integer library such as Boost multiprecision
+ * to give result any modular exponentation problem reletively quickly.
```

Comment:
```suggestion
 * to give result any modular exponentiation problem relatively quickly.
```

Reply:
I'll turn on spell check in vim. Is this repo using US or GB English?

Reply:
AFAIK, it's US English.